src/App.tsx

```diff
@@ -1,61 +1,152 @@
+import { useState } from 'react';
+
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+import { TodoList } from './components/TodoList';
+import { Todo } from './types/Todo';
+
 import './App.scss';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+const getUserById = (userId: number) => {
+  return usersFromServer.find(({ id }) => id === userId) || null;
+};
+
+const initialTodos = todosFromServer.map(todo => (
+  {
+    ...todo,
+    user: getUserById(todo.userId),
+  }
+));
+
+const isAppropriateChar = (ch: string) => {
+  const correctCharSet = [
+    'abcdefghijklmnopqrstuvwxyz',
+    'абвгдеёжзийклмнопрстуфхцчшщъыьэюя',
+    '0123456789',
+    ' ',
+  ];
+
+  return correctCharSet.some(charSet => charSet.includes(ch.toLowerCase()));
+};
```

Comment:
Let's use regex here, I found one that will help in this case

`^[a-z0-9\s]+$`